src/System.Threading/ref/System.Threading.cs

```diff
@@ -186,7 +186,7 @@ public static partial class LazyInitializer
     }
     public partial struct LockCookie
     {
-        private int _dummyPrimitive;
+        private object _dummy;
```

Comment:
Similarly here, not sure why this change was made:
https://github.com/dotnet/corefx/blob/16f3eef83d6df4c9dbc0e2992dafbc9553db363d/src/System.Threading/src/System/Threading/LockCookie.cs#L13-L17